docs/gitlab_integration.md

```diff
@@ -0,0 +1,132 @@
+information how to
+[register a GitLab Runner](https://docs.gitlab.com/runner/register/index.html).
+
+# 2. Define a job in your .gitlab-ci.yml
+GitLab CI/CD pipelines are configured using a YAML file called `.gitlab-ci.yml`
+within each project. Create this file in your project root directory:
+
+```yml
+code_quality:
+  allow_failure: false
+  script:
+    - bash .gitlab/ci/run_codechecker.sh
+  after_script:
+    - cat gl-code-quality-report.json
+  artifacts:
+    reports:
+      codequality: gl-code-quality-report.json
+    paths: [gl-code-quality-report.json]
+    expire_in: '5 yrs'
+  stage: test
+```
+
+`expire_in` allows you to specify how long artifacts should live before they
+expire and are therefore deleted. You can set it to a lower or a higher value.
+For more information
+[see](https://docs.gitlab.com/ee/ci/yaml/README.html#artifactsexpire_in).
+
+Script is a shell script which will be executed by the Runner.
+
+# 3. Create script to run CodeChecker analysis
+Create a shell script in your project under `.gitlab/ci/run_codechecker.sh`.
+This script will run the analysis and it has to create a json file called
+`gl-code-quality-report.json` which will contains reports in Code Climate
+format. CodeChecker from version `6.12.0` is able to generate Code Climate
+output by using the *CodeChecker parse* or *CodeChecker cmd diff* commands.
+
+```sh
+#!/usr/bin/env bash
+
+# Log your project.
+CodeChecker log -b "make" -o ./compilation_database.json
+
+# Analyze your project.
+CodeChecker analyze --ctu \
+  -o ./reports \
+  ./compilation_database.json
+
+# Create the report file by using the CodeChecker parse command.
+CodeChecker parse \
+  --trim-path-prefix $(pwd) \
+  -e codeclimate
+  ./reports > gl-code-quality-report.json
+
+# Or you can create a report by using the CodeChecker diff command.
+CC_REPO_DIR="$(pwd)" \
+CodeChecker cmd diff \
+  -b my_remote_run \
+  --url your_baseline_url \
+  -n ./reports \
+  --new \
+  -o codeclimate \
+  -e ./codeclimate && \
+cp codeclimate/codeclimate_issues.json gl-code-quality-report.json
+
+# Exit with status code 1 if there is any report in the output file.
+status=$(cat gl-code-quality-report.json)
+if [[ -n "$status" && "$status" != "[]" ]]; then
+  exit 1
+fi
+```
+
+4. Create a merge request
+Create a merge which will contain the `.gitlab-ci.yml` and the
+`.gitlab/ci/run_codechecker.sh` files.
+
+Once the Code Quality job has completed, potential changes to code quality are
+shown directly in the merge request. If the Code Quality report doesn’t have
+anything to compare to, no information will be displayed in the merge request
+area. You may see the following message: *Failed to load codeclimate report*.
+That is the case when you add the Code Quality job in your `.gitlab-ci.yml`
+for the very first time.
+
+For more information
+[see](https://docs.gitlab.com/ee/user/project/merge_requests/code_quality.html).
+
+# Possible errors
```

Comment:
Is there a link to the official documentation to handle these errors, if there is some changes there this section could get outdated.